### PR TITLE
CI: add tests for windows

### PR DESCRIPTION
**Pull request by pascalgrimaud**
Fix https://github.com/jhipster/jhipster-lite/issues/302

### CODE REVIEW

**Comment by pascalgrimaud** [comment]
@pblanchardie : the CI for Windows is ready. I'll merge it as soon as all the tests with Windows are fixed -> https://github.com/jhipster/jhipster-lite/issues/438

**Comment by DanielFran** [comment]
@pascalgrimaud The tests are also occuring when I compile on windows...

**Comment by pascalgrimaud** [comment]
Can you put the log here plz @DanielFran ?

**Comment by DanielFran** [comment]
`[INFO] --- maven-surefire-plugin:3.0.0-M5:test (default-test) @ jhlite ---
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running tech.jhipster.lite.AnnotationArchTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 3.946 s - in tech.jhipster.lite.AnnotationArchTest
[INFO] Running tech.jhipster.lite.HexagonalArchTest
[ERROR] Tests run: 10, Failures: 1, Errors: 9, Skipped: 0, Time elapsed: 0.504 s <<< FAILURE! - in tech.jhipster.lite.HexagonalArchTest
[ERROR] tech.jhipster.lite.HexagonalArchTest.shouldNotHaveInfrastructureDependenciesFromDomain  Time elapsed: 0.001 s  <<< FAILURE!
java.lang.AssertionError
	at tech.jhipster.lite.HexagonalArchTest.lambda$toPackage$6(HexagonalArchTest.java:73)
	at tech.jhipster.lite.HexagonalArchTest.getPackageAnnotatedBy(HexagonalArchTest.java:55)
	at tech.jhipster.lite.HexagonalArchTest.buildBusinessContexts(HexagonalArchTest.java:37)
	at tech.jhipster.lite.HexagonalArchTest.<clinit>(HexagonalArchTest.java:30)

[ERROR] tech.jhipster.lite.HexagonalArchTest.shouldNotDependOnOtherContexts  Time elapsed: 0.001 s  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class tech.jhipster.lite.HexagonalArchTest

[ERROR] tech.jhipster.lite.HexagonalArchTest.domainShouldNotInteractWithOutside  Time elapsed: 0.001 s  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class tech.jhipster.lite.HexagonalArchTest

[ERROR] tech.jhipster.lite.HexagonalArchTest.shouldNotHaveInfrastructureDependenciesInApplication  Time elapsed: 0.001 s  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class tech.jhipster.lite.HexagonalArchTest

[ERROR] tech.jhipster.lite.HexagonalArchTest.applicationShouldNotInteractWithSecondary  Time elapsed: 0 s  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class tech.jhipster.lite.HexagonalArchTest

[ERROR] tech.jhipster.lite.HexagonalArchTest.shouldNotHaveContextDependenciesForSharedKernels  Time elapsed: 0 s  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class tech.jhipster.lite.HexagonalArchTest

[ERROR] tech.jhipster.lite.HexagonalArchTest.primaryShouldNotInteractWithSecondary  Time elapsed: 0 s  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class tech.jhipster.lite.HexagonalArchTest

[ERROR] tech.jhipster.lite.HexagonalArchTest.shouldNotHaveFrameworkDependenciesInContext  Time elapsed: 0 s  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class tech.jhipster.lite.HexagonalArchTest

[ERROR] tech.jhipster.lite.HexagonalArchTest.shouldHaveNoDependencyToApplicationFromSecondary  Time elapsed: 0 s  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class tech.jhipster.lite.HexagonalArchTest

[ERROR] tech.jhipster.lite.HexagonalArchTest.shouldNotHavePublicControllers  Time elapsed: 0 s  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class tech.jhipster.lite.HexagonalArchTest

[INFO] Running tech.jhipster.lite.common.domain.Base64UtilsTest
[ERROR] OpenJDK 64-Bit Server VM warning: Sharing is only supported for boot loader classes because bootstrap classpath has been appended
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.243 s - in tech.jhipster.lite.common.domain.Base64UtilsTest
[INFO] Running tech.jhipster.lite.common.domain.FileUtilsTest
[INFO] Running tech.jhipster.lite.common.domain.FileUtilsTest$ContainsInLineTest
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.024 s - in tech.jhipster.lite.common.domain.FileUtilsTest$ContainsInLineTest
[INFO] Running tech.jhipster.lite.common.domain.FileUtilsTest$ContainsLinesTest
[INFO] Tests run: 7, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.265 s - in tech.jhipster.lite.common.domain.FileUtilsTest$ContainsLinesTest
[INFO] Running tech.jhipster.lite.common.domain.FileUtilsTest$ContainsRegexpTest
[INFO] Tests run: 8, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.017 s - in tech.jhipster.lite.common.domain.FileUtilsTest$ContainsRegexpTest
[INFO] Running tech.jhipster.lite.common.domain.FileUtilsTest$CountsRegexpTest
[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.02 s - in tech.jhipster.lite.common.domain.FileUtilsTest$CountsRegexpTest
[INFO] Running tech.jhipster.lite.common.domain.FileUtilsTest$CreateFolderTest
[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.567 s - in tech.jhipster.lite.common.domain.FileUtilsTest$CreateFolderTest
[INFO] Running tech.jhipster.lite.common.domain.FileUtilsTest$DetectEndOfLineTest
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.019 s - in tech.jhipster.lite.common.domain.FileUtilsTest$DetectEndOfLineTest
[INFO] Running tech.jhipster.lite.common.domain.FileUtilsTest$ExistsTest
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.006 s - in tech.jhipster.lite.common.domain.FileUtilsTest$ExistsTest
[INFO] Running tech.jhipster.lite.common.domain.FileUtilsTest$FileSystemTest
[WARNING] Tests run: 2, Failures: 0, Errors: 0, Skipped: 1, Time elapsed: 0.007 s - in tech.jhipster.lite.common.domain.FileUtilsTest$FileSystemTest
[INFO] Running tech.jhipster.lite.common.domain.FileUtilsTest$GetLineTest
[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.016 s - in tech.jhipster.lite.common.domain.FileUtilsTest$GetLineTest
[INFO] Running tech.jhipster.lite.common.domain.FileUtilsTest$GetPathTest
[WARNING] Tests run: 6, Failures: 0, Errors: 0, Skipped: 1, Time elapsed: 0.033 s - in tech.jhipster.lite.common.domain.FileUtilsTest$GetPathTest
[INFO] Running tech.jhipster.lite.common.domain.FileUtilsTest$ReadTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.005 s - in tech.jhipster.lite.common.domain.FileUtilsTest$ReadTest
[INFO] Running tech.jhipster.lite.common.domain.FileUtilsTest$ReplaceTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.005 s - in tech.jhipster.lite.common.domain.FileUtilsTest$ReplaceTest
[INFO] Tests run: 0, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.042 s - in tech.jhipster.lite.common.domain.FileUtilsTest
[INFO] Running tech.jhipster.lite.common.domain.OSUtilsTest
[WARNING] Tests run: 2, Failures: 0, Errors: 0, Skipped: 1, Time elapsed: 0.006 s - in tech.jhipster.lite.common.domain.OSUtilsTest
[INFO] Running tech.jhipster.lite.common.domain.WordUtilsTest
[INFO] Running tech.jhipster.lite.common.domain.WordUtilsTest$IdentWithSpaceTest
[INFO] Tests run: 6, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.019 s - in tech.jhipster.lite.common.domain.WordUtilsTest$IdentWithSpaceTest
[INFO] Running tech.jhipster.lite.common.domain.WordUtilsTest$IndentTest
[INFO] Tests run: 5, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.007 s - in tech.jhipster.lite.common.domain.WordUtilsTest$IndentTest
[INFO] Running tech.jhipster.lite.common.domain.WordUtilsTest$KebabCaseTest
[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.03 s - in tech.jhipster.lite.common.domain.WordUtilsTest$KebabCaseTest
[INFO] Running tech.jhipster.lite.common.domain.WordUtilsTest$UpperFirstTest
[INFO] Tests run: 5, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.007 s - in tech.jhipster.lite.common.domain.WordUtilsTest$UpperFirstTest
[INFO] Tests run: 0, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.072 s - in tech.jhipster.lite.common.domain.WordUtilsTest
[INFO] Running tech.jhipster.lite.error.domain.AssertTest
[INFO] Tests run: 21, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.04 s - in tech.jhipster.lite.error.domain.AssertTest
[INFO] Running tech.jhipster.lite.error.domain.GeneratorExceptionTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.002 s - in tech.jhipster.lite.error.domain.GeneratorExceptionTest
[INFO] Running tech.jhipster.lite.error.domain.MissingMandatoryValueExceptionTest
[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.008 s - in tech.jhipster.lite.error.domain.MissingMandatoryValueExceptionTest
[INFO] Running tech.jhipster.lite.error.domain.UnauthorizedValueExceptionTest
[INFO] Tests run: 9, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.011 s - in tech.jhipster.lite.error.domain.UnauthorizedValueExceptionTest
[INFO] Running tech.jhipster.lite.generator.buildtool.generic.domain.BuildToolDomainServiceTest
[INFO] Running tech.jhipster.lite.generator.buildtool.generic.domain.BuildToolDomainServiceTest$GradleTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.255 s - in tech.jhipster.lite.generator.buildtool.generic.domain.BuildToolDomainServiceTest$GradleTest
[INFO] Running tech.jhipster.lite.generator.buildtool.generic.domain.BuildToolDomainServiceTest$MavenTest
[INFO] Tests run: 9, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.099 s - in tech.jhipster.lite.generator.buildtool.generic.domain.BuildToolDomainServiceTest$MavenTest
[INFO] Running tech.jhipster.lite.generator.buildtool.generic.domain.BuildToolDomainServiceTest$NoBuildToolTest
[INFO] Tests run: 7, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.021 s - in tech.jhipster.lite.generator.buildtool.generic.domain.BuildToolDomainServiceTest$NoBuildToolTest
[INFO] Tests run: 0, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.392 s - in tech.jhipster.lite.generator.buildtool.generic.domain.BuildToolDomainServiceTest
[INFO] Running tech.jhipster.lite.generator.buildtool.generic.domain.DependencyTest
[INFO] Tests run: 10, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.01 s - in tech.jhipster.lite.generator.buildtool.generic.domain.DependencyTest
[INFO] Running tech.jhipster.lite.generator.buildtool.generic.domain.ParentTest
[INFO] Tests run: 7, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.007 s - in tech.jhipster.lite.generator.buildtool.generic.domain.ParentTest
[INFO] Running tech.jhipster.lite.generator.buildtool.generic.domain.PluginTest
[INFO] Tests run: 9, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.008 s - in tech.jhipster.lite.generator.buildtool.generic.domain.PluginTest
[INFO] Running tech.jhipster.lite.generator.buildtool.maven.domain.maven.MavenDomainServiceTest
[INFO] Tests run: 12, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.146 s - in tech.jhipster.lite.generator.buildtool.maven.domain.maven.MavenDomainServiceTest
[INFO] Running tech.jhipster.lite.generator.buildtool.maven.domain.maven.MavenTest
[INFO] Tests run: 13, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.01 s - in tech.jhipster.lite.generator.buildtool.maven.domain.maven.MavenTest
[INFO] Running tech.jhipster.lite.generator.init.domain.InitDomainServiceTest
[INFO] Tests run: 6, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.022 s - in tech.jhipster.lite.generator.init.domain.InitDomainServiceTest
[INFO] Running tech.jhipster.lite.generator.init.domain.InitTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.001 s - in tech.jhipster.lite.generator.init.domain.InitTest
[INFO] Running tech.jhipster.lite.generator.packagemanager.npm.domain.NpmDomainServiceTest
[INFO] Tests run: 5, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.044 s - in tech.jhipster.lite.generator.packagemanager.npm.domain.NpmDomainServiceTest
[INFO] Running tech.jhipster.lite.generator.packagemanager.npm.domain.NpmTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.037 s - in tech.jhipster.lite.generator.packagemanager.npm.domain.NpmTest
[INFO] Running tech.jhipster.lite.generator.packagemanager.npm.infrastructure.secondary.NpmLocalRepositoryTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.155 s - in tech.jhipster.lite.generator.packagemanager.npm.infrastructure.secondary.NpmLocalRepositoryTest
[INFO] Running tech.jhipster.lite.generator.project.domain.CheckConfigTest
[INFO] Tests run: 10, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.023 s - in tech.jhipster.lite.generator.project.domain.CheckConfigTest
[INFO] Running tech.jhipster.lite.generator.project.domain.DefaultConfigTest
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.005 s - in tech.jhipster.lite.generator.project.domain.DefaultConfigTest
[INFO] Running tech.jhipster.lite.generator.project.domain.ProjectTest
[INFO] Running tech.jhipster.lite.generator.project.domain.ProjectTest$AddConfigTest
[INFO] Tests run: 6, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.016 s - in tech.jhipster.lite.generator.project.domain.ProjectTest$AddConfigTest
[INFO] Running tech.jhipster.lite.generator.project.domain.ProjectTest$BuildTest
[INFO] Tests run: 6, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.027 s - in tech.jhipster.lite.generator.project.domain.ProjectTest$BuildTest
[INFO] Running tech.jhipster.lite.generator.project.domain.ProjectTest$DetectEndOfLineTest
[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.021 s - in tech.jhipster.lite.generator.project.domain.ProjectTest$DetectEndOfLineTest
[INFO] Running tech.jhipster.lite.generator.project.domain.ProjectTest$GetConfigTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.002 s - in tech.jhipster.lite.generator.project.domain.ProjectTest$GetConfigTest
[INFO] Running tech.jhipster.lite.generator.project.domain.ProjectTest$GetIntegerConfigTest
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.005 s - in tech.jhipster.lite.generator.project.domain.ProjectTest$GetIntegerConfigTest
[INFO] Running tech.jhipster.lite.generator.project.domain.ProjectTest$GetStringConfigTest
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.005 s - in tech.jhipster.lite.generator.project.domain.ProjectTest$GetStringConfigTest
[INFO] Running tech.jhipster.lite.generator.project.domain.ProjectTest$PackageNameTest
[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.006 s - in tech.jhipster.lite.generator.project.domain.ProjectTest$PackageNameTest
[INFO] Tests run: 9, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.133 s - in tech.jhipster.lite.generator.project.domain.ProjectTest
[INFO] Running tech.jhipster.lite.generator.project.infrastructure.primary.dto.ProjectDTOTest
[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.008 s - in tech.jhipster.lite.generator.project.infrastructure.primary.dto.ProjectDTOTest
[INFO] Running tech.jhipster.lite.generator.project.infrastructure.secondary.GitUtilsTest
[ERROR] Tests run: 4, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.777 s <<< FAILURE! - in tech.jhipster.lite.generator.project.infrastructure.secondary.GitUtilsTest
[ERROR] tech.jhipster.lite.generator.project.infrastructure.secondary.GitUtilsTest.shouldApply  Time elapsed: 0.077 s  <<< ERROR!
org.eclipse.jgit.errors.InvalidObjectIdException: Invalid id: 483dde2
	at tech.jhipster.lite.generator.project.infrastructure.secondary.GitUtilsTest.shouldApply(GitUtilsTest.java:49)
Caused by: java.lang.ArrayIndexOutOfBoundsException
	at tech.jhipster.lite.generator.project.infrastructure.secondary.GitUtilsTest.shouldApply(GitUtilsTest.java:49)

[INFO] Running tech.jhipster.lite.generator.project.infrastructure.secondary.MustacheUtilsTest
[ERROR] Tests run: 8, Failures: 3, Errors: 0, Skipped: 0, Time elapsed: 0.073 s <<< FAILURE! - in tech.jhipster.lite.generator.project.infrastructure.secondary.MustacheUtilsTest
[ERROR] tech.jhipster.lite.generator.project.infrastructure.secondary.MustacheUtilsTest.shouldTemplateWithMap  Time elapsed: 0.055 s  <<< FAILURE!
org.opentest4j.AssertionFailedError: 

expected: 
  "The apero can start when there is some chips
"
 but was: 
  "The apero can start when there is some chips
  "
	at tech.jhipster.lite.generator.project.infrastructure.secondary.MustacheUtilsTest.shouldTemplateWithMap(MustacheUtilsTest.java:43)

[ERROR] tech.jhipster.lite.generator.project.infrastructure.secondary.MustacheUtilsTest.shouldTemplateInClassPath  Time elapsed: 0.008 s  <<< FAILURE!
org.opentest4j.AssertionFailedError: 

expected: 
  "The apero can start when there is some chips
"
 but was: 
  "The apero can start when there is some chips
  "
	at tech.jhipster.lite.generator.project.infrastructure.secondary.MustacheUtilsTest.shouldTemplateInClassPath(MustacheUtilsTest.java:33)

[ERROR] tech.jhipster.lite.generator.project.infrastructure.secondary.MustacheUtilsTest.shouldTemplate  Time elapsed: 0.002 s  <<< FAILURE!
org.opentest4j.AssertionFailedError: 

expected: 
  "The apero can start when there is some chips
"
 but was: 
  "The apero can start when there is some chips
  "
	at tech.jhipster.lite.generator.project.infrastructure.secondary.MustacheUtilsTest.shouldTemplate(MustacheUtilsTest.java:23)

[INFO] Running tech.jhipster.lite.generator.project.infrastructure.secondary.ProjectLocalRepositoryTest
[ERROR] Tests run: 30, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 1.166 s <<< FAILURE! - in tech.jhipster.lite.generator.project.infrastructure.secondary.ProjectLocalRepositoryTest
[ERROR] tech.jhipster.lite.generator.project.infrastructure.secondary.ProjectLocalRepositoryTest.shouldGitApplyPatch  Time elapsed: 0.027 s  <<< ERROR!
org.eclipse.jgit.errors.InvalidObjectIdException: Invalid id: 483dde2
	at tech.jhipster.lite.generator.project.infrastructure.secondary.ProjectLocalRepositoryTest.shouldGitApplyPatch(ProjectLocalRepositoryTest.java:338)
Caused by: java.lang.ArrayIndexOutOfBoundsException
	at tech.jhipster.lite.generator.project.infrastructure.secondary.ProjectLocalRepositoryTest.shouldGitApplyPatch(ProjectLocalRepositoryTest.java:338)

[INFO] Running tech.jhipster.lite.generator.server.javatool.base.domain.JavaBaseDomainServiceTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.005 s - in tech.jhipster.lite.generator.server.javatool.base.domain.JavaBaseDomainServiceTest
[INFO] Running tech.jhipster.lite.generator.server.javatool.jacoco.domain.JacocoDomainServiceTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.004 s - in tech.jhipster.lite.generator.server.javatool.jacoco.domain.JacocoDomainServiceTest
[INFO] Running tech.jhipster.lite.generator.server.sonar.domain.SonarTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.001 s - in tech.jhipster.lite.generator.server.sonar.domain.SonarTest
[INFO] Running tech.jhipster.lite.generator.server.springboot.aop.logging.domain.maven.AopLoggingDomainServiceTest
[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.079 s - in tech.jhipster.lite.generator.server.springboot.aop.logging.domain.maven.AopLoggingDomainServiceTest
[INFO] Running tech.jhipster.lite.generator.server.springboot.banner.domain.BannerDomainServiceTest
[INFO] Tests run: 5, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.009 s - in tech.jhipster.lite.generator.server.springboot.banner.domain.BannerDomainServiceTest
[INFO] Running tech.jhipster.lite.generator.server.springboot.core.domain.SpringBootDomainServiceTest
[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.023 s - in tech.jhipster.lite.generator.server.springboot.core.domain.SpringBootDomainServiceTest
[INFO] Running tech.jhipster.lite.generator.server.springboot.core.domain.SpringBootTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.001 s - in tech.jhipster.lite.generator.server.springboot.core.domain.SpringBootTest
[INFO] Running tech.jhipster.lite.generator.server.springboot.database.mysql.domain.MySQLTest
[INFO] Tests run: 5, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.004 s - in tech.jhipster.lite.generator.server.springboot.database.mysql.domain.MySQLTest
[INFO] Running tech.jhipster.lite.generator.server.springboot.database.postgresql.domain.PostgresqlTest
[INFO] Tests run: 6, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.007 s - in tech.jhipster.lite.generator.server.springboot.database.postgresql.domain.PostgresqlTest
[INFO] Running tech.jhipster.lite.generator.server.springboot.logging.domain.SpringBootLoggingDomainServiceTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.02 s - in tech.jhipster.lite.generator.server.springboot.logging.domain.SpringBootLoggingDomainServiceTest
[INFO] Running tech.jhipster.lite.generator.server.springboot.logstash.tcp.domain.LogstashTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.002 s - in tech.jhipster.lite.generator.server.springboot.logstash.tcp.domain.LogstashTest
[INFO] Running tech.jhipster.lite.generator.server.springboot.mvc.security.jwt.domain.JwtSecurityDomainServiceTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.039 s - in tech.jhipster.lite.generator.server.springboot.mvc.security.jwt.domain.JwtSecurityDomainServiceTest
[INFO] Running tech.jhipster.lite.generator.server.springboot.mvc.security.jwt.domain.JwtSecurityTest
[INFO] Tests run: 8, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.012 s - in tech.jhipster.lite.generator.server.springboot.mvc.security.jwt.domain.JwtSecurityTest
[INFO] Running tech.jhipster.lite.generator.server.springboot.mvc.web.domain.SpringBootMvcTest
[INFO] Tests run: 7, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.006 s - in tech.jhipster.lite.generator.server.springboot.mvc.web.domain.SpringBootMvcTest
[INFO] Running tech.jhipster.lite.generator.server.springboot.properties.domain.SpringBootPropertiesDomainServiceTest
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.022 s - in tech.jhipster.lite.generator.server.springboot.properties.domain.SpringBootPropertiesDomainServiceTest
[INFO] Running tech.jhipster.lite.generator.server.springboot.springcloud.configclient.domain.SpringCloudConfigTest
[INFO] Tests run: 5, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.005 s - in tech.jhipster.lite.generator.server.springboot.springcloud.configclient.domain.SpringCloudConfigTest
[INFO] Running tech.jhipster.lite.technical.infrastructure.primary.exception.BadRequestAlertExceptionTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.021 s - in tech.jhipster.lite.technical.infrastructure.primary.exception.BadRequestAlertExceptionTest
[INFO] Running tech.jhipster.lite.technical.infrastructure.primary.exception.FieldErrorDTOTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.006 s - in tech.jhipster.lite.technical.infrastructure.primary.exception.FieldErrorDTOTest
[INFO] Running tech.jhipster.lite.technical.infrastructure.primary.exception.HeaderUtilTest
[INFO] Tests run: 10, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.048 s - in tech.jhipster.lite.technical.infrastructure.primary.exception.HeaderUtilTest
[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   HexagonalArchTest.<clinit>:30->buildBusinessContexts:37->getPackageAnnotatedBy:55->lambda$toPackage$6:73
[ERROR]   MustacheUtilsTest.shouldTemplate:23 
expected: 
  "The apero can start when there is some chips
"
 but was: 
  "The apero can start when there is some chips
  "
[ERROR]   MustacheUtilsTest.shouldTemplateInClassPath:33 
expected: 
  "The apero can start when there is some chips
"
 but was: 
  "The apero can start when there is some chips
  "
[ERROR]   MustacheUtilsTest.shouldTemplateWithMap:43 
expected: 
  "The apero can start when there is some chips
"
 but was: 
  "The apero can start when there is some chips
  "
[ERROR] Errors: 
[ERROR]   HexagonalArchTest.applicationShouldNotInteractWithSecondary » NoClassDefFound ...
[ERROR]   HexagonalArchTest.domainShouldNotInteractWithOutside » NoClassDefFound Could n...
[ERROR]   HexagonalArchTest.primaryShouldNotInteractWithSecondary » NoClassDefFound Coul...
[ERROR]   HexagonalArchTest.shouldHaveNoDependencyToApplicationFromSecondary » NoClassDefFound
[ERROR]   HexagonalArchTest.shouldNotDependOnOtherContexts » NoClassDefFound Could not i...
[ERROR]   HexagonalArchTest.shouldNotHaveContextDependenciesForSharedKernels » NoClassDefFound
[ERROR]   HexagonalArchTest.shouldNotHaveFrameworkDependenciesInContext » NoClassDefFound
[ERROR]   HexagonalArchTest.shouldNotHaveInfrastructureDependenciesInApplication » NoClassDefFound
[ERROR]   HexagonalArchTest.shouldNotHavePublicControllers » NoClassDefFound Could not i...
[ERROR]   GitUtilsTest.shouldApply:49 » InvalidObjectId Invalid id: 483dde2
[ERROR]   ProjectLocalRepositoryTest.shouldGitApplyPatch:338 » InvalidObjectId Invalid i...
[INFO] 
[ERROR] Tests run: 368, Failures: 4, Errors: 11, Skipped: 3
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  20.002 s
[INFO] Finished at: 2022-01-07T18:38:22Z
[INFO] ------------------------------------------------------------------------
`

**Comment by pascalgrimaud** [comment]
@DanielFran : 1 test left and it should be ok :-)

**Comment by pascalgrimaud** [comment]
@pblanchardie @DanielFran : I managed to find a computer with Windows and tested my branch. With all the fix here, it works well, so I don't understand why the CI failed.
Can you try my branch with your computer plz?

**Comment by DanielFran** [comment]
@pascalgrimaud Will test it this afternoon...

File: src/test/java/tech/jhipster/lite/generator/project/infrastructure/secondary/ProjectLocalRepositoryTest.java [hunk 1]
```diff
@@ -191,17 +192,18 @@ void shouldNotContainsRegexpWithoutExistingFile() {
   @Test
   void shouldReplaceText() {
     Project project = tmpProjectWithPomXml();
```
Comment (pblanchardie) [inline src/test/java/tech/jhipster/lite/generator/project/infrastructure/secondary/ProjectLocalRepositoryTest.java:194]: @pascalgrimaud the project is created with default to LF, then the CRLF pom is added. You have to create the project with endOfLine=CRLF for this test to pass.

Comment (pblanchardie) [reply src/test/java/tech/jhipster/lite/generator/project/infrastructure/secondary/ProjectLocalRepositoryTest.java:194]: you can verify by replacing this line with:
```java
    Project initProject = tmpProjectWithPomXml();
    Project project = Project.builder().folder(initProject.getFolder()).build();
```

Comment (pascalgrimaud) [reply src/test/java/tech/jhipster/lite/generator/project/infrastructure/secondary/ProjectLocalRepositoryTest.java]: still not working :(

File: src/main/java/tech/jhipster/lite/generator/project/infrastructure/secondary/ProjectLocalRepository.java [hunk 2]
```diff
@@ -126,7 +127,7 @@ public void write(Project project, String text, String destination, String desti
 
     try {
       FileUtils.createFolder(projectDestination);
-      Files.write(getPathOf(projectDestinationFilename), text.getBytes());
+      Files.write(getPathOf(projectDestinationFilename), text.replace(WordUtils.LF, project.getEndOfLine()).getBytes());
```
Comment (pblanchardie) [inline src/main/java/tech/jhipster/lite/generator/project/infrastructure/secondary/ProjectLocalRepository.java:130]: revert to `Files.write(getPathOf(projectDestinationFilename), text.getBytes());` and it works
or leave replacements as \n always
or only replace \n if it there is not \r

File: src/test/java/tech/jhipster/lite/generator/project/infrastructure/secondary/ProjectLocalRepositoryTest.java [hunk 3]
```diff
@@ -190,18 +191,21 @@ void shouldNotContainsRegexpWithoutExistingFile() {
...
       <name>jhipster</name>
-        <description>JHipster Project</description>""";
-    String newText = """
+        <description>JHipster Project</description>""".replace(WordUtils.LF, project.getEndOfLine());
+    String newText =
+      """
       <name>chips</name>
-
-        <description>Chips Project</description>""";
+        <description>Chips Project</description>""".replace(WordUtils.LF, project.getEndOfLine());
```
Comment (pblanchardie) [inline src/test/java/tech/jhipster/lite/generator/project/infrastructure/secondary/ProjectLocalRepositoryTest.java:204]: replacement is already crlf

**Comment by pascalgrimaud** [comment]
- closing in favor of https://github.com/jhipster/jhipster-lite/pull/509

**Pull request closed without merging**